  * Make sure selected .a files in the current build directory are appended to
    the build path.
  * run ranlib on all .a files so they actually link correctly

--- ext/extmk.rb
+++ ext/extmk.rb
@@ -550,6 +550,7 @@ void Init_ext _((void))\n{\n#$extinit}
   end
 
   $extobjs = "ext/#{extinit.o} #{$extobjs}"
+  $extobjs = "#{$extobjs} libz.a libcrypto.a libssl.a"
   if RUBY_PLATFORM =~ /beos/
     $extflags.delete("-L/usr/local/lib")
   end
@@ -563,10 +564,19 @@ void Init_ext _((void))\n{\n#$extinit}
   ].map {|n, v|
     "#{n}=#{v}" if v and !(v = v.strip).empty?
   }.compact
-  puts(*conf)
+  # puts(*conf)
   $stdout.flush
   $mflags.concat(conf)
   $makeflags.concat(conf)
+  unless enable_config("shared", $enable_shared )
+    $extobjs.split.each do |eo|
+      if eo =~/^ext/ and eo =~ /\.a$/ then
+        local_f = eo[4..-1]    
+        puts "ranlib #{local_f}"
+        %x[ ranlib #{local_f} ]
+      end
+    end
+  end
 else
   FileUtils.rm_f(extinit.to_a)
 end
diff --git a/ext/openssl/openssl_missing.h b/ext/openssl/openssl_missing.h